compiler/qsc_rca/src/applications.rs

```diff
@@ -0,0 +1,405 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::{
+    common::{
+        aggregate_compute_kind, aggregate_value_kind, initalize_locals_map, InputParam,
+        InputParamIndex, Local, LocalKind, LocalsLookup,
+    },
+    scaffolding::PackageScaffolding,
+    ApplicationsTable, ComputeKind, QuantumProperties, RuntimeFeatureFlags, ValueKind,
+};
+use qsc_data_structures::index_map::IndexMap;
+use qsc_fir::fir::{BlockId, ExprId, NodeId, Pat, PatId, PatKind, SpecDecl, StmtId};
+use rustc_hash::FxHashMap;
+
+#[derive(Debug)]
+pub struct ApplicationInstancesTable {
+    pub inherent: ApplicationInstance,
+    pub dynamic_params: Vec<ApplicationInstance>,
+    is_settled: bool,
```

Comment:
What does `is_settled` mean? From the code it looks like a flag indicating that the object has been fully populated.

Reply:
That's exactly what it meant. After refactoring, the flag is no longer needed.